include/podio/detail/LinkObj.h

```diff
@@ -0,0 +1,53 @@
+#ifndef PODIO_DETAIL_LINKOBJ_H
+#define PODIO_DETAIL_LINKOBJ_H
+
+#include "podio/detail/LinkFwd.h"
+
+#include "podio/ObjectID.h"
+
+namespace podio {
+
+template <typename FromT, typename ToT>
+class LinkObj {
+
+  friend Link<FromT, ToT>;
+  friend MutableLink<FromT, ToT>;
+
+public:
+  /// Constructor
+  LinkObj() : id(), data(LinkData{1.0f}), m_from(nullptr), m_to(nullptr) {
+  }
+
+  /// Constructor from ObjectID and data (does not initialize relations yet!)
+  LinkObj(const podio::ObjectID id_, LinkData data_) : id(id_), data(data_) {
+  }
+
+  /// Copy constructor (deep-copy of relations)
+  LinkObj(const LinkObj& other) : id(), data(other.data), m_from(nullptr), m_to(nullptr) {
+    if (other.m_from) {
+      m_from = new FromT(*other.m_from);
+    }
+    if (other.m_to) {
+      m_to = new ToT(*other.m_to);
+    }
+  }
+
+  /// No assignment operator
+  LinkObj& operator=(const LinkObj&) = delete;
+
+  /// Destructor
+  ~LinkObj() {
+    delete m_from;
+    delete m_to;
+  }
+
+public:
+  podio::ObjectID id{};
+  LinkData data{1.0f};
+  FromT* m_from{nullptr};
```

Comment:
do we want to use some smart pointer here instead of bare ones?

Reply:
These pointers don't manage anything. Additionally, this is also what "equivalent" generated code would look like, so they can stay as bare ones.

Reply:
OK. Fine. If that's the only delete I am fine.

Reply:
Ah wait, I think I spoke to soon. Maybe these should be `unique_ptr`s after all, since they are indeed manually `delete`d. I was mixing things up with the `Obj*` that are inside these pointers.

Reply:
This now done with `unique_ptr`s. Since I need the `Link`s and other generated code to behave the same, I have made the changes for the generated classes in #689 (which is now also included in this PR, so that has to be merged first).